src/main/groovy/jenkins/automation/utils/ScmUtils.groovy

```diff
@@ -24,8 +32,9 @@ class ScmUtils {
                         branch("*/tags/$version")
                     }
                     relativeTargetDir(repo.sub_directory)
+                    shallowClone(shallow)
```

Comment:
Is this what you mean: 
```
        Boolean shallow = false`
        shallow = repo.shallow
```
But then shallow is null and then `shallowClone(shallow)` receives a null argument and throws an error.